Src/IronPython.Modules/_codecs.cs

```diff
@@ -182,41 +184,37 @@ public static PythonTuple escape_decode(CodeContext/*!*/ context, [NotNull]strin
-            var buf = new List<byte>(data.Length);
-            var span = data.Span;
+        public static PythonTuple/*!*/ escape_encode([NotNull]IBufferProtocol data, string? errors = null) {
+            using IPythonBuffer buffer = data.GetBuffer();
+            var result = new List<byte>(buffer.ItemCount);
+            var span = buffer.AsReadOnlySpan();
             for (int i = 0; i < span.Length; i++) {
                 byte b = span[i];
                 switch (b) {
-                    case (byte)'\n': buf.Add((byte)'\\'); buf.Add((byte)'n'); break;
-                    case (byte)'\r': buf.Add((byte)'\\'); buf.Add((byte)'r'); break;
-                    case (byte)'\t': buf.Add((byte)'\\'); buf.Add((byte)'t'); break;
-                    case (byte)'\\': buf.Add((byte)'\\'); buf.Add((byte)'\\'); break;
-                    case (byte)'\'': buf.Add((byte)'\\'); buf.Add((byte)'\''); break;
+                    case (byte)'\n': result.Add((byte)'\\'); result.Add((byte)'n'); break;
+                    case (byte)'\r': result.Add((byte)'\\'); result.Add((byte)'r'); break;
+                    case (byte)'\t': result.Add((byte)'\\'); result.Add((byte)'t'); break;
+                    case (byte)'\\': result.Add((byte)'\\'); result.Add((byte)'\\'); break;
+                    case (byte)'\'': result.Add((byte)'\\'); result.Add((byte)'\''); break;
                     default:
                         if (b < 0x20 || b >= 0x7f) {
-                            buf.AddRange($"\\x{b:x2}".Select(c => unchecked((byte)c)));
+                            result.AddRange($"\\x{b:x2}".Select(c => unchecked((byte)c)));
                         } else {
-                            buf.Add(b);
+                            result.Add(b);
                         }
                         break;
                 }
             }
-            return PythonTuple.MakeTuple(Bytes.Make(buf.ToArray()), data.Length);
+            return PythonTuple.MakeTuple(Bytes.Make(result.ToArray()), buffer.ItemCount);
         }
 
         #endregion
 
         #region Latin-1 Functions
 
-        private static PythonTuple latin_1_decode(CodeContext context, ReadOnlyMemory<byte> input, string? errors = null)  // TODO: remove
-            => DoDecode(context, "latin-1", Encoding.GetEncoding("iso-8859-1"), input, errors, input.Length).ToPythonTuple();
-
         public static PythonTuple latin_1_decode(CodeContext context, [NotNull]IBufferProtocol input, string? errors = null) {
-            // pilot interface change
-            using IPythonBuffer buf = input.GetBuffer();
-            ReadOnlyMemory<byte> mem = buf.ToMemory();
-            return DoDecode(context, "latin-1", Encoding.GetEncoding("iso-8859-1"), mem, errors, mem.Length).ToPythonTuple();
+            using IPythonBuffer buffer = input.GetBuffer();
+            return DoDecode(context, "latin-1", Encoding.GetEncoding("iso-8859-1"), buffer, errors, buffer.ItemCount).ToPythonTuple();
```

Comment:
Shouldn't this be the "bytes length" instead of `ItemCount`? Same comment everywhere `ItemCount` is used.
```Python
assert _codecs.latin_1_decode(array.array("I", (1633771873,))) == ('aaaa', 4)
```

Reply:
You're right, I cut corners here too much, wanted to save obtaining the span. In general, "bytes length" == `ItemCount * ItemSize`, but I'd rather not use this expression here.

Reply:
Whatever the solution, we should probably add a test for this anyway.